crypto/relic_build.sh

```diff
@@ -12,17 +12,48 @@ chmod -R +w "$(pwd)"
 mkdir -p "$DIR/relic/build"
 pushd "$DIR/relic/build"
 
-# Set RELIC config for Flow
 
-#
+# make cmake print its CC interpretation
+echo $DIR
+CMAKE_FILE="${DIR}/relic/CMakeLists.txt"
+CMAKE_PRINT_CC="message ( STATUS \"CC=\$ENV{CC}\" )"
+# Make the cmake run print its interpretation of CC
+echo ${CMAKE_PRINT_CC} >> ${CMAKE_FILE}
+
+# Probe cmake's MakeFile generation and extract the CC version
+CMAKE_OUTPUT=$(mktemp)
+cmake .. > $CMAKE_OUTPUT
+CC_VAL="$(tail -n 5 "$CMAKE_OUTPUT" | grep -oE -m 1 'CC=.*$')"
+CC_VAL="${CC_VAL:3}"
+
+# de-mangle the CMakeLists file, using a temporary file for BSD compatibility
+CMAKE_TEMP=$(mktemp)
```

Comment:
[very minor and optional] we could define `CMAKE_TEMP` at line 24 and avoid declaring `CMAKE_OUTPUT`